### ISSUE TITLE
Synchronized charts with different width gutters become desynced

### ISSUE THREAD

**Issue by Bilge** [apexcharts/apexcharts.js#1404]
If the y-axis gutters are different widths between two synchronized charts, the charts will be slightly desynchronised by however much difference there is in gutter widths.

![image](https://user-images.githubusercontent.com/470626/76914630-c9f58680-68b2-11ea-8075-6fe7ea9dac57.png)

Synchronization should track the x-axis data point rather than pixel offsets.

**Comment by junedchhipa** [comment]
This is a known issue which is mentioned on the [docs page](https://apexcharts.com/docs/chart-types/synchronized-charts/)

The workaround is to have the same width y-axis for all synced charts.

**Comment by Bilge** [comment]
Just because it's noted does not mean it's a solution, right? It's a good thing it's acknowledged, but I think this issue should remain open until such a time as a solution is in place to address it.

**Comment by junedchhipa** [comment]
Sorry that you felt offended by my last comment, I just noticed my tone wasn't right.
The reason I have left it on the user to set the y-axis width is that I have already tried setting the y-axis width for all synced charts internally. That didn't go really well after many attempts.

I will re-visit this part once I think the major issues/bugs of the library are fixed. As this is already on the docs, it will be on my mind to think about the solutions for it.

Thanks.

**Comment by Bilge** [comment]
I wasn't offended, I just think this issue should not be closed until the issue is resolved, don't you agree?

I don't think the solution lies in trying to match y-axis widths. I think it should be a requirement that both charts have the same x-axis data points, and if that's the case, they can be synchronized on their values rather than their pixel offsets.

**Comment by junedchhipa** [comment]
You're right. I will think about an alternative way as you described. Thanks for the input.

**Comment by Bilge** [comment]
@junedchhipa I made a [CodePen](https://codepen.io/bilge-the-encoder/pen/yLYLMZo) to help you diagnose this issue because it's important to me so I want to make it as easy as possible for you to fix :)

**Comment by Fnizou** [comment]
Hello, I have the same problem with two types of graph, it would really be necessary that the synchronization be done by the value in X and not by the positioning in px, it will be great to find a solution, similar, I am stuck in my project, Thanks to you,

![bug](https://user-images.githubusercontent.com/64423991/80388547-5039bb00-88aa-11ea-9ae4-21cd50f3f6ad.jpg)

**Comment by Bilge** [comment]
@junedchhipa Is this something that can be fixed without too much difficulty? It's still a major problem.

**Comment by Fnizou** [comment]
in the meantime maybe do a hack in css :
`class="apexcharts-inner apexcharts-graphical" transform="translate(33.125, 30)"
`
the concern is on the x of the transform, it must be constrained in the same way on the two graphs,

Maybe also :
```
.apexcharts-yaxis{
min-width:150px;
overflow:hidden
} 
```

it's just a track,
good luck

**Comment by Bilge** [comment]
@Fnizou Any hack with absolute values is ***invalid*** because each chart has different data with different scales and different gutter widths.

**Comment by Fnizou** [comment]
yes it's true, 
you can make a JS loop on all the graphs, and apply the widest width to them ...
But @junedchhipa will fix it soon ;)

**Comment by caiosantanaj** [comment]
Hi!
I have the same issue! I'm trying to do a synchronized chart with a multi-axis and a single axis. A better way to sync size will be very welcomed!
I fixed my problem putting a second axis in my single chart and changed the color of the label to the same color of the background
![Chart](https://i.imgur.com/iRP38AI.png)

**Comment by JaMinsane** [comment]
@caiosantanaj Nice idea. Can you share some code? I'm pretty confused how change a color of the second Y-axis of the second chart. I hope you can help me. Thanks!

**Comment by caiosantanaj** [comment]
Sorry for the delay! 
I hope this can help: https://codesandbox.io/s/currying-water-ucreb?file=/src/example.js

Just search for "#f00" or HERE in the sandbox

**Comment by Bilge** [comment]
I have abandoned this charting library at this point.

**Comment by graingerkid** [comment]
Any update on this?

**Comment by graingerkid** [comment]
I've set the following on each chart and it's still not synced up - is this not what you were meaning?

```
      yaxis: {
    labels: {
      minWidth: 40
    }
  },
```

**Comment by gabrielbunselmeyer** [comment]
I get that this is stale, but 3 years after and from what I can see it's still an issue.

Has there been any progress on it?

**Comment by Uzair16ahmed** [comment]
I currently have two charts and have given them their ids and a common group name. The issue i am facing is the options of second chart are overlapping the first chart like shown in the picture. the first chart have 8 different series and with their individual color but second chart yellow color overlaps them. can someone explain this?
<img width="445" alt="image" src="https://github.com/user-attachments/assets/50381e48-648e-4a34-a31a-9ea90d90f757">
